crates/next-core/src/next_font_google/util.rs

```diff
@@ -247,22 +249,36 @@ pub(crate) fn extract_font_urls(
                     .map(|m| m.as_str());
 
                 if let Some(url) = font_url {
-                    if !declarations.iter().any(|d| d.url == url) {
-                        declarations.push(FontResource {
-                            url: url.to_owned(),
-                            should_preload: should_preload
-                                && subsets.contains(&current_subset.context(
-                                    "Invariant: subset should be set by preceeding comment at \
-                                     this point",
-                                )?),
-                        })
+                    if !all_urls.contains(url) {
+                        all_urls.insert(url.to_owned());
```

Comment:
Nit: I think these two statements can be joined, `insert()` should return a boolean.

Reply:
The second statement allocates but the first doesn't, so it's not 100% equivalent.